agnpy/tests/test_sedintegrable.py

```diff
@@ -0,0 +1,82 @@
+import numpy as np
+import math
+import astropy.units as u
+from astropy.constants import m_e, h
+from astropy.coordinates import Distance
+from agnpy.spectra import PowerLaw
+from agnpy.emission_regions import Blob
+from agnpy.synchrotron import Synchrotron
+from ..utils.sedintegrable import SedFluxIntegrable
+
+
+def _estimate_powerlaw_integral(x_values, y_values):
+    slope, intercept = np.polyfit(np.log10(x_values.value), np.log10(y_values.value), 1)
+    power = slope + 1
+    estimated_integral = (10 ** intercept) * (
+            x_values[-1].value ** power / power - x_values[0].value ** power / power)
+    return estimated_integral
+
+
+def _sample_synchrotron_model():
+    # Based on the synchrotron example from agnpy documentation
+    R_b = 1e16 * u.cm
+    n_e = PowerLaw.from_total_energy(1e48 * u.erg, 4 / 3 * np.pi * R_b ** 3, p=2.8, gamma_min=1e2,
+                                     gamma_max=1e7, mass=m_e)
+    blob = Blob(R_b, z=Distance(1e27, unit=u.cm).z, delta_D=10, Gamma=10, B=1 * u.G, n_e=n_e)
+    synch = Synchrotron(blob)
+    return synch
+
+
+class TestSedIntegrable:
+
+    def test_flat_integral(self):
+        """Integrate over flat SED (Fnu equal to 1.0 for all frequencies)"""
+        class FlatSedGenerator(SedFluxIntegrable):
+            def sed_flux(self, nu):
+                return np.full(fill_value=1.0, shape=nu.shape, dtype=np.float64) * (u.erg / (u.cm ** 2 * u.s * u.Hz)) * nu
+
+        nu_min = 10 * u.Hz
+        nu_max = 10 ** 20 * u.Hz
+        flux = FlatSedGenerator().integrate_flux(nu_min, nu_max)
+        assert flux == 1e+20 * u.erg / (u.cm ** 2 * u.s)
+
+
+    def test_synchrotron_energy_flux_integral(self):
+        """Integrate over sample synchrotron flux and compare with the value estimated using the power law integral"""
+        synch = _sample_synchrotron_model()
+
+        nu_min_log = 13
+        nu_max_log = 20
+        nu_min = 10 ** nu_min_log * u.Hz
+        nu_max = 10 ** nu_max_log * u.Hz
+
+        nu = np.logspace(nu_min_log, nu_max_log) * u.Hz
+        Fnu = synch.sed_flux(nu) / nu
+        estimated_integral = _estimate_powerlaw_integral(nu, Fnu)
+
+        actual_integral = synch.integrate_flux(nu_min, nu_max).value
+
+        assert math.isclose(actual_integral, estimated_integral, rel_tol=0.05)
+
+    def test_synchrotron_photon_flux_integral(self):
+        """Integrate over sample synchrotron photon flux and compare with the value estimated using the power law integral"""
+        synch = _sample_synchrotron_model()
+
+        nu_min_log = 13
+        nu_max_log = 20
+        nu_min = 10 ** nu_min_log * u.Hz
+        nu_max = 10 ** nu_max_log * u.Hz
+
+        nu = np.logspace(nu_min_log, nu_max_log) * u.Hz
+        Fnu = synch.sed_flux(nu) / nu
+        # convert the energy flux to photon flux in photons / s / cm2 / eV
+        photons_per_Hz = Fnu / nu.to("erg", equivalencies=u.spectral())  # Fnu is in ergs so must use the same unit
```

Comment:
Improve a bit the unit usage, remember that you can always convert the ratio to the unit you like with the `.to` method. Valid also for the next two lines, why do you need to re-define `h` to get the energies, when you could directly use the `spectral_equivalencies` method?

But see my previous point here? If we had a general `SedFlux` handling all these conversions, then we could handle more easily these conversions within the tests.

Reply:
I remember i had some troubles with this conversions, for simple things the automatic spectral conversion worked fine, but for ratios etc it didn't. Maybe i just didn't know how to use it correctly?
For example if I try to convert photon flux per frequency (photons / cm2 Hz s), to photon flux per energy (photons / cm2 eV s), the code like this:
`photons_per_Hz.to("1/(cm2 eV s)", equivalencies=u.spectral())`
fails with the error: 
`astropy.units.core.UnitConversionError: 
'1 / (cm2 Hz s)' (column density) and '1 / (cm2 eV s)' are not convertible`